File: sortingAlgo/insertionSort/insertionSort.py
# Testcase
arr = [1, 63, 103, 3, 8, 74, 76, 37, 71, 66]
l = len(arr)

def insertionSort(arr):
	for i in range(1,l):
		if arr[i] < arr[i - 1]:
			for j in range(i):
				if arr[i] < arr[j]:
					x = arr[i]
					del arr[i]
					arr.insert(j, x)
	return arr

print(insertionSort(arr))

File: sortingAlgo/insertionSort/insertionSort.cc
#include<iostream>

using namespace std;

void insertionsort(int a[], int n) {
    for(int i = 1; i < n; ++i) {
        int j = i-1;
        int key = a[i];
        while(j > -1 && a[j] > key) {
            swap(a[j+1], a[j]);
            --j;
        }
        a[j+1] = key;
    }
}

void insertionsort_rec(int a[], int n) {
    if(n < 2) return;
    insertionsort_rec(a, n-1);
 
    int key = a[n-1];
    int j = n-2;

    while(j > -1 && a[j] > key) {
        swap(a[j+1], a[j]);
        --j;
    }

    a[j+1] = key;

}

int main() {
    int arr[10] = {3, 450, 12, 4, -1, 0, 24, 95, 123, 0};
    for(int i = 0; i < 10; ++i)
        cout << *(arr+i) << ' ';
    cout << endl;
    insertionsort_rec(arr, 10);
    for(int i = 0; i < 10; ++i)
        cout << *(arr+i) << ' ';
    return 0;
}
